dds/DCPS/DiscoveryBase.h

```diff
@@ -1263,6 +1200,253 @@ namespace OpenDDS {
         }
       }
 
+      void
+      remove_expired_endpoints(const DCPS::MonotonicTimePoint& /*now*/)
+      {
+        MatchingDataIter it;
+        for (it = matching_data_buffer.begin(); it != matching_data_buffer.end(); it++) {
+          if (MonotonicTimePoint::now() - it->second.time_added_to_map >= max_reply_period) {
+            // TLS_TODO: sanity check on locks
```

Comment:
A lock on the `matching_data_buffer` would probably be a good idea.

Reply:
Implemented - needs verification

Reply:
I think that's probably good.